modules/auxiliary/scanner/rdp/cve_2019_0708_bluekeep.rb

```diff
@@ -0,0 +1,565 @@
+    coop = rdp_encrypted_pkt(["1a001700f103ea03010000010c00140000000400000000000000"].pack("H*"), rc4enckey, hmackey)
+    vprint_status("Grea2t!")
+    rdp_send(synch + coop)
+
+    vprint_status("Sending client control request control PDU")
+    rdp_send(rdp_encrypted_pkt(["1a001700f103ea03010000010c00140000000100000000000000"].pack("H*"), rc4enckey, hmackey))
+
+    vprint_status("Sending client persistent key list PDU")
+    rdp_send(rdp_encrypted_pkt(pdu_client_persistent_key_list(), rc4enckey, hmackey))
+
+    vprint_status("Sending client font list PDU")
+    rdp_send(rdp_encrypted_pkt(["1a001700f103ea03010000010c00270000000000000003003200"].pack("H*"), rc4enckey, hmackey))
+
+    #vprint_status("Sending base PDU")
+    #rdp_send(rdp_encrypted_pkt(["030000001d0002000308002004051001400a000c840000000000000000590d381001cc"].pack("H*"), rc4enckey, hmackey))
+
+
+
+    #res = rdp_recv()
+    #vprint_good("#{bin_to_hex(res)}")
+
+    result = try_check(rc4enckey, hmackey)
+
+
+
+    if result == Exploit::CheckCode::Vulnerable
+      report_goods
+    end
+
+
+    # Can't determine, but at least I know the service is running
+    return result
+  end
+
+  def check_host(ip)
+    # The check command will call this method instead of run_host
+
+    status = Exploit::CheckCode::Unknown
+
+    begin
+      begin
+        connect
+      rescue ::Errno::ETIMEDOUT, Rex::HostUnreachable, Rex::ConnectionTimeout, Rex::ConnectionRefused, ::Timeout::Error, ::EOFError => e
+        return Exploit::CheckCode::Unsupported	 # used to display custom msg error
+      end
+
+      status = Exploit::CheckCode::Detected
+
+      sock.setsockopt(::Socket::IPPROTO_TCP, ::Socket::TCP_NODELAY, 1)
+      status = check_rdp_vuln
+    rescue Rex::AddressInUse, ::Errno::ETIMEDOUT, Rex::HostUnreachable, Rex::ConnectionTimeout, Rex::ConnectionRefused, ::Timeout::Error, ::EOFError, ::TypeError => e
+      bt = e.backtrace.join("\n")
+      vprint_error("Unexpected error: #{e.message}")
+      vprint_line(bt)
+      elog("#{e.message}\n#{bt}")
+    rescue RdpCommunicationError => e
+      vprint_error("Error communicating RDP protocol.")
+      status = Exploit::CheckCode::Unknown
+    rescue Errno::ECONNRESET => e # NLA?
+      vprint_error("Connection reset, possible NLA is enabled.")
+    rescue => e
+      bt = e.backtrace.join("\n")
+      vprint_error("Unexpected error: #{e.message}")
+      vprint_line(bt)
+      elog("#{e.message}\n#{bt}")
+    ensure
+      disconnect
+    end
+
+    status
+  end
+
+  def run_host(ip)
+    # Allow the run command to call the check command
+
+    status = check_host(ip)
+    if status == Exploit::CheckCode::Vulnerable
+      print_good("#{status[1]}")
+    elsif status == Exploit::CheckCode::Unsupported	 # used to display custom msg error
+      print_status("The target service is not running, or refused our connection.")
+    else
+      print_status("#{status[1]}")
+    end
+  end
```

Comment:
Since you already have `status`, I would return it at the end of this method so we can hook this module into the future exploit with minimal effort.

Reply:
added return to check_ip